### PR TITLE
add goaisatsu keyboard.

### PR DESCRIPTION
**Pull request by okuura**
## Description
Add JIS base 82 switches PCB. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/okuura/goaisatsu/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+    [3] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (dunk2k) [inline keyboards/okuura/goaisatsu/keymaps/via/keymap.c:53]: removing non-functional layers. VIA defaults to 4 layers.
```suggestion
    )
```

File: keyboards/okuura/goaisatsu/info.json [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                {"matrix":[11,5], "x":14.25, "y":5.25},
+                {"matrix":[11,6], "x":15.25, "y":5.25}
+            ]
+        }
+    },
+    "split": {
+        "enabled": true,
+        "transport": {
+            "protocol": "serial"
+        },
```
Comment (dunk2k) [inline keyboards/okuura/goaisatsu/info.json:118]: not required/default
```suggestion
```

File: keyboards/okuura/goaisatsu/info.json [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "matrix_pins": {
+        "cols": ["F4", "F5", "F6", "F7", "B1", "B3", "B2", "B6"],
+        "rows": ["D4", "C6", "D7", "E6", "B4", "B5"]
+    },
+    "processor": "atmega32u4",
```
Comment (dunk2k) [inline keyboards/okuura/goaisatsu/info.json:19]: pro micro is used (mentioned in `readme.md`)
```suggestion
    "development_board": "promicro",
    "diode_direction": "COL2ROW",
    "features": {
        "bootmagic": true,
        "command": false,
        "console": false,
        "extrakey": true,
        "mousekey": true,
        "nkro": true
    },
    "matrix_pins": {
        "cols": ["F4", "F5", "F6", "F7", "B1", "B3", "B2", "B6"],
        "rows": ["D4", "C6", "D7", "E6", "B4", "B5"]
    },
```

File: keyboards/okuura/goaisatsu/info.json [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                {"matrix":[5,3], "x":3.25, "y":5.25, "w":1.25},
+                {"matrix":[5,4], "x":4.5, "y":5.25, "w":1.25},
+                {"matrix":[5,5], "x":5.75, "y":5.25},
+                {"matrix":[11,0], "x":8.75, "y":5.25},
+                {"matrix":[11,1], "x":9.75, "y":5.25, "w":1.25},
+                {"matrix":[11,2], "x":11, "y":5.25, "w":1.25},
+                {"matrix":[11,3], "x":12.25, "y":5.25},
+                {"matrix":[11,4], "x":13.25, "y":5.25},
+                {"matrix":[11,5], "x":14.25, "y":5.25},
+                {"matrix":[11,6], "x":15.25, "y":5.25}
```
Comment (dunk2k) [inline keyboards/okuura/goaisatsu/info.json:110]: very simple formatting that should appease formatting "enthusiasts" 😉 
```suggestion
                {"matrix": [0, 0], "x": 0, "y": 0},
                {"matrix": [0, 1], "x": 1.25, "y": 0},
                {"matrix": [0, 2], "x": 2.25, "y": 0},
                {"matrix": [0, 3], "x": 3.25, "y": 0},
                {"matrix": [0, 4], "x": 4.25, "y": 0},
                {"matrix": [0, 5], "x": 5.5, "y": 0},
                {"matrix": [6, 0], "x": 8.5, "y": 0},
                {"matrix": [6, 1], "x": 9.5, "y": 0},
                {"matrix": [6, 2], "x": 10.5, "y": 0},
                {"matrix": [6, 3], "x": 11.75, "y": 0},
                {"matrix": [6, 4], "x": 12.75, "y": 0},
                {"matrix": [6, 5], "x": 13.75, "y": 0},
                {"matrix": [6, 6], "x": 14.75, "y": 0},
                {"matrix": [6, 7], "x": 15.75, "y": 0},
                
                {"matrix": [1, 0], "x": 0, "y": 1.25},
                {"matrix": [1, 1], "x": 1, "y": 1.25},
                {"matrix": [1, 2], "x": 2, "y": 1.25},
                {"matrix": [1, 3], "x": 3, "y": 1.25},
                {"matrix": [1, 4], "x": 4, "y": 1.25},
                {"matrix": [1, 5], "x": 5, "y": 1.25},
                {"matrix": [1, 6], "x": 6, "y": 1.25},
                {"matrix": [7, 0], "x": 9, "y": 1.25},
                {"matrix": [7, 1], "x": 10, "y": 1.25},
                {"matrix": [7, 2], "x": 11, "y": 1.25},
                {"matrix": [7, 3], "x": 12, "y": 1.25},
                {"matrix": [7, 4], "x": 13, "y": 1.25},
                {"matrix": [7, 5], "x": 14, "y": 1.25},
                {"matrix": [7, 6], "x": 15, "y": 1.25},
                {"matrix": [7, 7], "x": 16, "y": 1.25},
                
                {"matrix": [2, 0], "x": 0, "y": 2.25, "w": 1.5},
                {"matrix": [2, 1], "x": 1.5, "y": 2.25},
                {"matrix": [2, 2], "x": 2.5, "y": 2.25},
                {"matrix": [2, 3], "x": 3.5, "y": 2.25},
                {"matrix": [2, 4], "x": 4.5, "y": 2.25},
                {"matrix": [2, 5], "x": 5.5, "y": 2.25},
                {"matrix": [8, 0], "x": 8.5, "y": 2.25},
                {"matrix": [8, 1], "x": 9.5, "y": 2.25},
                {"matrix": [8, 2], "x": 10.5, "y": 2.25},
                {"matrix": [8, 3], "x": 11.5, "y": 2.25},
                {"matrix": [8, 4], "x": 12.5, "y": 2.25},
                {"matrix": [8, 5], "x": 13.5, "y": 2.25},
                {"matrix": [8, 6], "x": 14.5, "y": 2.25},
                {"matrix": [8, 7], "x": 15.75, "y": 2.25, "w": 1.25, "h": 2},
                
                {"matrix": [3, 0], "x": 0, "y": 3.25, "w": 1.75},
                {"matrix": [3, 1], "x": 1.75, "y": 3.25},
                {"matrix": [3, 2], "x": 2.75, "y": 3.25},
                {"matrix": [3, 3], "x": 3.75, "y": 3.25},
                {"matrix": [3, 4], "x": 4.75, "y": 3.25},
                {"matrix": [3, 5], "x": 5.75, "y": 3.25},
                {"matrix": [9, 0], "x": 8.75, "y": 3.25},
                {"matrix": [9, 1], "x": 9.75, "y": 3.25},
                {"matrix": [9, 2], "x": 10.75, "y": 3.25},
                {"matrix": [9, 3], "x": 11.75, "y": 3.25},
                {"matrix": [9, 4], "x": 12.75, "y": 3.25},
                {"matrix": [9, 5], "x": 13.75, "y": 3.25},
                {"matrix": [9, 6], "x": 14.75, "y": 3.25},
                
                {"matrix": [4, 0], "x": 0, "y": 4.25, "w": 2.25},
                {"matrix": [4, 1], "x": 2.25, "y": 4.25},
                {"matrix": [4, 2], "x": 3.25, "y": 4.25},
                {"matrix": [4, 3], "x": 4.25, "y": 4.25},
                {"matrix": [4, 4], "x": 5.25, "y": 4.25},
                {"matrix": [4, 5], "x": 6.25, "y": 4.25},
                {"matrix": [10, 0], "x": 9.25, "y": 4.25},
                {"matrix": [10, 1], "x": 10.25, "y": 4.25},
                {"matrix": [10, 2], "x": 11.25, "y": 4.25},
                {"matrix": [10, 3], "x": 12.25, "y": 4.25},
                {"matrix": [10, 4], "x": 13.25, "y": 4.25},
                {"matrix": [10, 5], "x": 14.25, "y": 4.25},
                {"matrix": [10, 6], "x": 15.25, "y": 4.25},
                
                {"matrix": [5, 0], "x": 0, "y": 5.25, "w": 1.25},
                {"matrix": [5, 1], "x": 1.25, "y": 5.25},
                {"matrix": [5, 2], "x": 2.25, "y": 5.25},
                {"matrix": [5, 3], "x": 3.25, "y": 5.25, "w": 1.25},
                {"matrix": [5, 4], "x": 4.5, "y": 5.25, "w": 1.25},
                {"matrix": [5, 5], "x": 5.75, "y": 5.25},
                {"matrix": [11, 0], "x": 8.75, "y": 5.25},
                {"matrix": [11, 1], "x": 9.75, "y": 5.25, "w": 1.25},
                {"matrix": [11, 2], "x": 11, "y": 5.25, "w": 1.25},
                {"matrix": [11, 3], "x": 12.25, "y": 5.25},
                {"matrix": [11, 4], "x": 13.25, "y": 5.25},
                {"matrix": [11, 5], "x": 14.25, "y": 5.25},
                {"matrix": [11, 6], "x": 15.25, "y": 5.25}
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/okuura/goaisatsu/readme.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+# goaisatsu
+
+![goaisatsu](https://s2.booth.pm/7d7df4a1-8345-407f-9c2f-8eefe1f7a17f/i/4980888/1d7d5d1b-0eda-4db5-af38-a39bece6c12e_base_resized.jpg)
```
Comment (dunk2k) [inline keyboards/okuura/goaisatsu/readme.md:3]: could this image please be re-hosted on imgur, also including the `h` before .jpeg/.png in url  
e.g. `https://i.imgur.com/SK0flNs.jpeg` -> `https://i.imgur.com/SK0flNsh.jpeg`

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/okuura/goaisatsu/readme.md (hunk 5 above)
Comment (okuura) [reply keyboards/okuura/goaisatsu/readme.md]: Thank you for all your suggestions😀 You have made this PR better.
link has also been changed to imgur.
https://i.imgur.com/rd0MFFlh.jpg

**Review by dunk2k — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/okuura/goaisatsu/readme.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+* Hardware Supported: goaisatsuPCB, ProMicro
+* Hardware Availability: [Booth](https://okuura.booth.pm/items/4980888)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make goaisatsu:default
+
+Flashing example for this keyboard:
+
+    make goaisatsu:default:flash
```
Comment (drashna) [inline keyboards/okuura/goaisatsu/readme.md:17]: Could you add instructions on how to enter the bootloader, to flash new firmware?  This is needed, as a new user may not be aware of how to enter to bootloader to flash new firmware.

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `QK_BOOT` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/okuura/goaisatsu/readme.md (hunk 6 above)
Comment (okuura) [reply keyboards/okuura/goaisatsu/readme.md:17]: @drashna Thanks for your review. I added a description of the bootloader.

**Review by waffle87 — APPROVED** [review]

File: keyboards/okuura/goaisatsu/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
```
Comment (zvecr) [inline keyboards/okuura/goaisatsu/keymaps/default/keymap.c:16]:
```suggestion
// Copyright 2023 okuura (@okuura)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/okuura/goaisatsu/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
```
Comment (zvecr) [inline keyboards/okuura/goaisatsu/keymaps/via/keymap.c:16]:
```suggestion
// Copyright 2023 okuura (@okuura)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/okuura/goaisatsu/readme.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
```
Comment (zvecr) [inline keyboards/okuura/goaisatsu/readme.md:17]:
```suggestion
    make okuura/goaisatsu:default

Flashing example for this keyboard:

    make okuura/goaisatsu:default:flash
```

File: keyboards/okuura/goaisatsu/config.h [hunk 10]
```diff
@@ -0,0 +1,7 @@
+// Copyright 2023 okuura (@okuura)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#define TAPPING_TERM 150
```
Comment (zvecr) [inline keyboards/okuura/goaisatsu/config.h:6]: Setting this at the keyboard level should be avoided
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request closed without merging**